src/core/distributions/phylogenetics/substitution/AbstractPhyloCTMCSiteHomogeneous.h

```diff
@@ -2340,30 +2343,22 @@ void RevBayesCore::AbstractPhyloCTMCSiteHomogeneous<charType>::keepSpecializatio
     // reset flags for likelihood computation
     touched = false;
 
-    // reset the ln probability
-    this->storedLnProb = this->lnProb;
+    // we don't have a previous state anymore.
+    this->storedLnProb = {};
 
     // reset all flags
-    for (std::vector<bool>::iterator it = this->dirty_nodes.begin(); it != this->dirty_nodes.end(); ++it)
-    {
-        (*it) = false;
-    }
```

Comment:
Stop marking all partial likelihoods as up-to-date.

Reply:
The current values in dirty_nodes should already be right, we shouldn't change them all to `false`.